src/s3/types.rs

```diff
@@ -1482,7 +1482,10 @@ pub struct Destination {
 impl Destination {
     pub fn from_xml(element: &Element) -> Result<Destination, Error> {
         Ok(Destination {
-            bucket_arn: get_text(element, "Bucket")?,
+            bucket_arn: element
+                .get_child("Destination")
+                .ok_or_else(|| Error::XmlError("<Destination> tag not found".into()))
+                .and_then(|element| get_text(element, "Bucket"))?,
```

Comment:
You should use current logic in `1628` not here.

Reply:
Reverted.